src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs

```diff
@@ -54,8 +54,18 @@ public string GetPropertyValue(string name)
 
         protected async Task<ProjectInstance> BuildAsync(string taskName, MSB.Framework.ITaskHost taskHost, CancellationToken cancellationToken)
         {
-            // prepare for building
-            var buildTargets = new BuildTargets(_loadedProject, "Compile");
+            // Create a project instance to be executed by build engine.
+            // The executed project will hold the final model of the project after execution via msbuild.
+            var executedProject = _loadedProject.CreateProjectInstance();
+
+            // if the project doesn't have a Compile target, then its not really a vb or c# project.
+            if (!executedProject.Targets.ContainsKey("Compile"))
+            {
+                return executedProject;
+            }
+
+            // prepare for building  -- use CoreBuild to get xaml generated files
+            var buildTargets = new BuildTargets(_loadedProject, "CoreBuild");
```

Comment:
@KevinH-MS mentioned that you can add a `TestAssemblyName.dll.config` file, and xunit will apply it's contents when running tests for that assembly.  Does that work here?